tests/test_delete_default_cloudtrail/main.tf

```
@@ -0,0 +1,100 @@
+locals {
+  cloudtrail_name_prefix = "cloudtrail-${local.id}-"
+  cloudtrail_suffix      = "test-deletion"
+
+  id      = random_string.id.result
+  project = "test-delete-ct-${local.id}"
+
+  tags = {
+    "project" = local.project
+  }
+}
+
+module "test_cloudtrail" {
+  source             = "./cloudtrail"
+  cloudtrail_name    = "${local.cloudtrail_name_prefix}${local.cloudtrail_suffix}"
+  cloudtrail_s3_name = "${local.cloudtrail_name_prefix}${local.cloudtrail_suffix}-bucket"
+  tags               = local.tags
+}
```

Ahh, localstack only supports cloudtrail in the "Pro" version. Just _creating_ the lambda, we don't need the cloudtrail. I'd suggest adding a variable to the test, `create_cloudtrail` with a default value of `false`, and condition this module using `count`. That way we can run the mockstack test successfully. And if we want to test locally and exercise the lambda more thoroughly, we can pass `-var create_cloudtrail=true`...

```suggestion
module "test_cloudtrail" {
  count = var.create_cloudtrail ? 1 : 0

  source             = "./cloudtrail"
  cloudtrail_name    = "${local.cloudtrail_name_prefix}${local.cloudtrail_suffix}"
  cloudtrail_s3_name = "${local.cloudtrail_name_prefix}${local.cloudtrail_suffix}-bucket"
  tags               = local.tags
}
```